src/main/java/uk/nhs/tis/sync/job/PostFundingSyncJob.java

```diff
@@ -0,0 +1,83 @@
+package uk.nhs.tis.sync.job;
+
+import com.transformuk.hee.tis.tcs.api.enumeration.Status;
+import com.transformuk.hee.tis.tcs.service.model.Post;
+import com.transformuk.hee.tis.tcs.service.model.PostFunding;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+/**
+ * Job for updating funding status for posts. This class extends the
+ * PersonDateChangeCaptureSyncJobTemplate and provides functionality to synchronize post funding
+ * status based on specified criteria.
+ */
+@Component
+@ManagedResource(objectName = "sync.mbean:name=PostFundingSyncJob",
+    description = "Job for updating funding status for posts")
+public class PostFundingSyncJob extends PostFundingSyncJobTemplate<Post> {
+
+  private static final String BASE_QUERY = " SELECT postId "
+      + "  FROM PostFunding "
+      + "      WHERE postId > :lastPostId "
+      + "      AND startDate IS NOT NULL "
+      + "      AND (endDate = ':endDate' OR endDate IS NULL) "
+      + " GROUP BY postId "
+      + " ORDER BY postId LIMIT :pageSize ";
```

Comment:
:+1:
This is much simpler and therefore easier to spot the inefficiencies (that we might not be able to work around), e.g. Any posts with 1+ Funding elements will get processed every day.